### PR TITLE
Modernization of TC codebase

### PR DESCRIPTION
**Pull request by prigoyal**
This PR makes following changes:

1. Create a top level 'tc' folder, create tc/proto and move .proto files there. They should be included as tc/proto/mapping_options.pb.h
2. Move subfolders of include/tc into tc/ and same for subfolders of src/

addresses https://github.com/facebookresearch/TensorComprehensions/issues/194

cc @mingzhe09088  (couldn't tag as reviewer hence cc'ing)

### CODE REVIEW

**Comment by prigoyal** [comment]
@salexspb, I will make the -Werror change in a separate PR since that also involves testing some configuration changes on CI. This PR is ready for review and the incremental change can be landed if you want.

**Comment by nicolasvasilache** [comment]
Re hotfix see #281

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ include_directories(.)
 include_directories(../include)
 
 add_subdirectory(lang)
-add_subdirectory(proto)
+#add_subdirectory(proto)
```
Comment (nicolasvasilache) [inline src/CMakeLists.txt]: dead code

File: tc/aten/aten_compiler.h [hunk 2]
```diff
@@ -22,9 +22,9 @@
 #include <ATen/ATen.h>
 #include <ATen/DLConvertor.h>
 
-#include "tc/aten/utils.h"
-#include "tc/core/execution_engine.h"
-#include "tc/lang/parser.h"
+#include "aten/utils.h"
+#include "core/execution_engine.h"
```
Comment (salexspb) [inline tc/aten/aten_compiler.h]: Actually having includes starting from "tc", in my opinion, is what we should be targeting. this way we provide a proper isolation between tc and any other libraries when TC is included as a submodule or is included in an FBCode fashion. 


Also number of changed files would be significantly smaller in this case :)

File: tc/core/cpu/cpu_mapping_options.h [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 #include <mapping_options.pb.h>
```
Comment (salexspb) [inline tc/core/cpu/cpu_mapping_options.h]: this one should be, IMO, tc/proto/mapping_options.pb , but we can do it in a separate PR, of course.

File: tc/aten/aten_compiler.h (hunk 2 above)
Comment (prigoyal) [reply tc/aten/aten_compiler.h]: I agree. I'll address this and update PR. There will be cmake change needed only for this then. Thanks.

**Comment by prigoyal** [comment]
I ran `test_python/run_test.sh` and `test.sh` both and they pass

@salexspb , if you could help with review to avoid more merge conflicts, I would really appreciate that. NOTE: the Jenkins CI is having downtime atm so running local test is best thing to do right now.

**Comment by prigoyal** [comment]
yep, I ran `test_python/run_test.sh` and `test.sh` both and they pass. Thanks @salexspb :)
